### DISCUSSION TITLE
1.0.4

### DISCUSSION THREAD

**Discussion by ilyhalight** [ilyhalight/voice-over-translation#3] [Обновления]
Если что-то не работает, либо вас не устраивает - пишите - постараюсь исправить, либо используйте старую версию веб-сервера

**Full Changelog**: https://github.com/ilyhalight/voice-over-translation/compare/1.0.3...1.0.4

<hr /><em>This discussion was created from the release <a href='https://github.com/ilyhalight/voice-over-translation/releases/tag/1.0.4'>1.0.4</a>.</em>

**Comment by crassher** [comment]
Спасибо за то,что продемонстрировали работу этого чуда!  Но установив его,есть один минус,по сравнению с яндексом. Голос перевода хорошо слышно(можно регулировать),но голос оригинала тоже громкий. А в яндексе пришглушается оригинал голоса.   Можно  это как то изменить?

**Reply by ilyhalight** [reply]
> 
Возможно, в будущем обновление добавлю пункт в меню, при активации которого будет автоматически снижаться громкость оригинального звука

**Comment by crassher** [comment]
класс. по быстрей бы по возможности. так то темка клёвая 🤗😀

вт, 6 сент. 2022 г. в 09:10, Toil ***@***.***>:

**Reply by alexey-marochkin** [reply]
Так то можно ползунком громкости самого ютюба приглушить)

**Comment by alexey-marochkin** [comment]
А API отрабатывает только youtube? Или Яндекс.Диск сможет?

**Reply by ilyhalight** [reply]
> А API отрабатывает только youtube? Или Яндекс.Диск сможет?

По словам ТП Яндекса, перевод поддерживается только на этих сайтах:
![изображение](https://user-images.githubusercontent.com/62353659/188862138-313a6e2c-69d9-4fc3-b9f6-b1d1905c7ddc.png)

Расширение на данный момент поддерживает только ютуб.

**Reply by alexey-marochkin** [reply]
Тут интересный вопрос на самом деле: Возможно ответ касается вылавливания браузером ссылок на поток от видеосервиса, а API при этом может отработать и любую ссылку которую засунуть в x-protobuf. Мне кажется, что API имеет функционал поддерживающий вытаскивание аудиопотока из различных видео форматов (контейнеров), а уж на вход получает ссылку на поток и неважно откуда эта ссылка главное чтобы доступна для API из интернета была. Если имеете тестовый стенд можете попробовать подпихнуть ссылку на видео лежащее на Яндекс.Диск?

**Reply by ilyhalight** [reply]
> Тут интересный вопрос на самом деле: Возможно ответ касается вылавливания браузером ссылок на поток от видеосервиса, а API при этом может отработать и любую ссылку которую засунуть в x-protobuf. Мне кажется, что API имеет функционал поддерживающий вытаскивание аудиопотока из различных видео форматов (контейнеров), а уж на вход получает ссылку на поток и неважно откуда эта ссылка главное чтобы доступна для API из интернета была. Если имеете тестовый стенд можете попробовать подпихнуть ссылку на видео лежащее на Яндекс.Диск?

К сожалению, не имею

**Comment by crassher** [comment]
блин вот я тупой  ))
сорян. надо просто убавить громкость на ютубе и всё!  ))))))

ср, 7 сент. 2022 г. в 10:05, Алексей Марочкин ***@***.***>:

**Comment by crassher** [comment]
Это конечно хорошо. Но так же видео можно скачать себе,залить его и слушать.
А вот в реальном времени ещё нет перевода онлайн? не вкурсе

ср, 7 сент. 2022 г. в 20:59, Toil ***@***.***>:

**Reply by ilyhalight** [reply]
> Это конечно хорошо. Но так же видео можно скачать себе,залить его и слушать. А вот в реальном времени ещё нет перевода онлайн? не вкурсе ср, 7 сент. 2022 г. в 20:59, Toil ***@***.***>:
> […](#)
> По словам ТП Яндекса, поддерживаются только эти сайты: [image: изображение] <https://user-images.githubusercontent.com/62353659/188862138-313a6e2c-69d9-4fc3-b9f6-b1d1905c7ddc.png> Расширение на данный момент поддерживает только ютуб. — Reply to this email directly, view it on GitHub <[#3 (reply in thread)](https://github.com/ilyhalight/voice-over-translation/discussions/3#discussioncomment-3583872)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AIOIMJMWMKW6NCJQJC626JTV5BYPTANCNFSM5TRM3X7A> . You are receiving this because you commented.Message ID: <ilyhalight/voice-over-translation/repo-discussions/3/comments/3583872@ github.com>

Перевод в реальном времени есть, но он очень ограниченный - поддерживает только некоторые каналы на ютубе:
https://yandex.ru/promo/browser/streams/m/01

**Reply by FiLinX** [reply]
Автор - а не знаешь, почему выводит вот это "Не удалось запросить перевод видео"  на этом ролике? - https://www.youtube.com/watch?v=3KEfnPm17KE
версия 1.0.8 
раньше вроде выводило всегда только "Перевод займёт больше минуты"  и сама кнопка под этой надписью оставалась активной, то есть подождав несколько секунд и повторно нажав уже на эту надпись - почти всегда срабатывал новый запрос и получение "Выполнено",  а сейчас при той "Не удалось .." кнопка уже не работает..

**Reply by ilyhalight** [reply]
> Автор - а не знаешь, почему выводит вот это "Не удалось запросить перевод видео" на этом ролике? - https://www.youtube.com/watch?v=3KEfnPm17KE версия 1.0.8 раньше вроде выводило всегда только "Перевод займёт больше минуты" и сама кнопка под этой надписью оставалась активной, то есть подождав несколько секунд и повторно нажав уже на эту надпись - почти всегда срабатывал новый запрос и получение "Выполнено", а сейчас при той "Не удалось .." кнопка уже не работает..

https://github.com/ilyhalight/voice-over-translation/issues/27

**Comment by crassher** [comment]
если вдруг ты добавишь фичу эту убавление, как узнать что она добавится?
предложит обновиться или автоматом сама обновится7

чт, 8 сент. 2022 г. в 00:42, Toil ***@***.***>:

**Comment by crassher** [comment]
Странно. на яндексе переводит,а в гугле ошибка.
К автору пожалуйста обратитесь

пт, 9 сент. 2022 г. в 01:36, Roman Vladimirovich F. <
***@***.***>:

**Reply by FiLinX** [reply]
упс.. извиняюсь, наверно я не туда написал..  не разбираюсь просто в этих вещах) но спасибо, создал отдельный вопрос..
и я не знал, что в самом яндекс браузере перевод на этом видео работает..

**Reply by FiLinX** [reply]
Эх..  всё оказалось трагичнее  -  расширение вообще перестало работать НА ЛЮБЫХ роликах с ютуба!  Везде теперь пишет "Не удалось запросить перевод видео", даже на тех, что ещё вчера смотрел с переводом через это расширение..
Видимо Яндекс что-то в апи поменял..  Жаль..

**Comment by crassher** [comment]
привет. чего то переводчик плохо работает.   или вообще не хочет
переводить.  проблемы?

пт, 9 сент. 2022 г. в 07:12, Roman Vladimirovich F. <
***@***.***>:

**Reply by FiLinX** [reply]
#27

**Comment by clubprofi** [comment]
Работал скрипт, а сейчас опять перестал переводить. Подскажите пожалуйста или дайте ссылку на видео, как установить версию со старыми серверами чтобы точно работало.

**Reply by ilyhalight** [reply]
Вариант для сельф-хоста (размещения сервера для перевода у себя на ПК):

1. Устанавливаете YandexBrowser и NodeJS v16.x LTS
2. В настройках YandexBrowser **отключаете работу в фоне**, а так же **автозапуск**
3. Скачиваете и распаковываете [архив](https://github.com/ilyhalight/voice-over-translation/releases/tag/1.0.5) с исходником версии 1.0.5 
4. Заполняете свои пути к YandexBrowser в конфиг-файле (он находится тут: "voice-over-translation/web-server/config/default.json")
5. Установите все модули необходимые для работы веб-сервера. Для этого, находясь в папке "web-server", пропишите в консоли: ```npm i``` и дождитесь установки всех модулей.
6. Запустите веб-сервер. Для этого, находясь в папке "web-server", пропишите в консоли: ```npm run start```
7.  Установите скрипт из скаченного архива

P.S.: Если хотите установить на VDS, то будет необходимо поменять айпи-адреса в скрипте на адреса VDS. С VDS на линуксе не будет работать из-за отсутствия окружения рабочего стола

**Reply by FiLinX** [reply]
@ilyhalight 
Похоже я чего-то недопонимаю, так как у меня этот выкрутас с сервером совсем не работает, и вообще хлючит))
Ситуация:
Имею вин10 LTSC 64, где дефолтным браузером назначен инсталлированный 32 битный srware_iron_87.0.4450.0 
Какое то время пользовал успешно 1.0.8 скрипт через обезьянку.
Сейчас вот решил попробовать эту версию с сервером, скачал и распаковал указанный архив с 1.0.5.
Скачал и инсталлировал Node v16.17.1 (LTS)
Кстати, а какую битность этого нода мне надо было ставить - под систему 64 или под браузеры которые у меня 32? я поставил 64
Дальше я в default.json прописал пути до имеющегося у меня яндексбрауза.
Опять же, я пользую репак-портабле сборку от Cento8 - http://effect8.ru/soft/browsers/yabrowser-portable.html
Версия ябрауза 22.9.1.950  - спецсборка для перевода видео - в самом низу там на странице.
И так это портабля, то пути у меня такие:
E:\YaBrows_videotrans\Browser\browser.exe
E:\YaBrows_videotrans\User Data
Ладно, дальше установил скрипт, перетащив его из папки в обезьянку на Ироне.
Затем выполнил npm i (подтянулась папка node_modules)
И за этим выполнил npm run start  - мелькнула консоль и всё..
В Ироне открываю ютубину иноземную  -  и ничего.. Никакой панели перевода нет от скрипта..
Ну стал проверять всё по шагам - увидел, что в default.json пути должны быть с двойными слешами))
Исправил, и на всякий случай даже перезагрузил систему..
Снова выполняю команду npm run start  -  и блин, кроме окна консоли запустился и сам яндексбрауз..  ыыыыы 
С чего он запустился то? Ведь всякие эти автозапуски и фоновые отключены - https://i.postimg.cc/Y9PCJjK3/screenshot-52.png
Да ещё и окно самой консоли тоже так и осталось висеть - https://i.postimg.cc/TYJpPdrR/screenshot-53.png
Ну, это хотя бы указало что сервер работает.
Иду в Ирон, запускаю буржуйскую ютубину  -  и НИХРЕНА блин, никакой панельки перевода..
Мдяя...  
И хрен его знает теперь, сколько пакости этот ябрауз успел натащить, запустившись не из под лаунчера с кучей предназначавшихся ключей отрубалок-заглушек..
Эхьь..

**Reply by ilyhalight** [reply]
В этой версии расширения, насколько я помню, еще не было совместимости с некоторыми расширениями для юзерскриптов (greasemonkey и др.)

На счёт консоли и открытого Яндекс браузера, к сожалению не вижу вариантов сделать их скрытыми, ибо в "headless" версии яндекс браузера не работает функция перевода видео.

На счет Nodejs, то желательно качать версию по разрядности винды.

Как будет время, обновлю актуальную версию скрипта для работы с веб-сервером.

**Reply by FiLinX** [reply]
Подожди, так это чего, запуск ябрауза, это так и подразумевалось чтоли? 
Непонял..
А зачем тогда вообще всё это? 
Ведь если ябрауз запущенный нужен, то проще же в своём дефолтном браузе (Ирон у меня) кликнуть правомышбю на ссылке буржуйвидоса и там на Open in external app (либо тупо ручками урл скопировать и в ябраузе вставить).
Вобщем, я чё то совсем нафиг непонимаю ничё.. Не, не моё это знач)

**Reply by JackieWaltRyan** [reply]
> А зачем тогда вообще всё это? Ведь если ябрауз запущенный нужен, то проще же в своём дефолтном браузе (Ирон у меня) кликнуть правомышбю на ссылке буржуйвидоса и там на Open in external app (либо тупо ручками урл скопировать и в ябраузе вставить). Вобщем, я чё то совсем нафиг непонимаю ничё.. Не, не моё это знач)

запускать сервер на своем же пк, это разве что из за безысходности... собственно как и запускать целый сервер только для одного пользователя...

а сама эта система делалась и рассчитана на то, что сам сервер, вместе с ябраузером и всем остальным барахлом, находится где то в дата центре, может даже в другой стране, или даже на другом континенте... и к одному этому серверу подключаются сразу сотни или даже тысячи пользователей...

**Comment by crassher** [comment]
походу всё,кина не будет.  надо ждать апдэйта.
на крипте есть желание  заработать стабильно ?

пт, 30 сент. 2022 г. в 20:09, Roman Vladimirovich F. <
***@***.***>:

**Reply by FiLinX** [reply]
@crassher а ты случаем не "НеВешайНос" а? )  
если это ты  -  то Огромное Спасибо тебе за поддержку ребят!
просто заработать на крипте, особенно как Он, ну только дурак не возжелает)

**Comment by crassher** [comment]
нет.    По крипте просто есть абуз на бонусах . Если станет
интересно,обращайтесь

сб, 1 окт. 2022 г. в 23:34, Roman Vladimirovich F. ***@***.***

**Comment by crassher** [comment]
щас попробовал. всё ок

ср, 7 сент. 2022 г. в 13:00, Boo ***@***.***>:
